### ISSUE TITLE
FinalClassCheck: False positive with anonymous classes

### ISSUE THREAD

**Issue by Vyom-Yadav** [checkstyle/checkstyle#11365]
I have read check documentation: https://checkstyle.sourceforge.io/config_design.html#FinalClass
I have downloaded the latest checkstyle from: https://checkstyle.org/cmdline.html#Download_and_Run
I have executed the cli and showed it below, as cli describes the problem better than 1,000 words

```bash
/var/tmp $ javac Test.java

/var/tmp $ cat config.xml
```
```xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
        "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
        "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
    <module name="TreeWalker">
        <module name="FinalClass"/>
    </module>
</module>
```
```
/var/tmp $ cat Test.java
```
```java
public class Test {

    class a { // expected no violation
        private a() {
        }
    }

    a obj = new a() {
    };
}
```
```bash
/var/tmp $ RUN_LOCALE="-Duser.language=en -Duser.country=US"
/var/tmp $ java $RUN_LOCALE -jar checkstyle-10.0-all.jar -c config.xml Test.java
Starting audit...
[ERROR] /var/tmp/Test.java:3:5: Class a should be declared as final. [FinalClass]
Audit done.
Checkstyle ends with 1 errors.
```
---

**Describe what you expect in detail.**
```bash
/var/tmp $ RUN_LOCALE="-Duser.language=en -Duser.country=US"
/var/tmp $ java $RUN_LOCALE -jar checkstyle-10.0-all.jar -c config.xml Test.java
Starting audit...
Audit done.
```
This was supposed to be fixed in #9357. Making the class final results in a compile-time error.

**Comment by Vyom-Yadav** [comment]
I am on it.

**Comment by romani** [comment]
fix is merged
